erpnext/templates/pages/integrations/authorizenet_checkout.html

```diff
@@ -45,16 +45,23 @@ <h2 class="text-center">{{ payment_context.description }}</h2>
 							<label>{{ _("Card Number") }}</label>
 						</div>
 						<div class="col-md-8 col-sm-6">
-							<input type="number" id="card-number" name="card-number" class="field"
-								placeholder="{{ _('XXXX-XXXX-XXXX-XXXX') }}" />
+							<input type="text" id="card-number" name="card-number" class="field" maxlength="16"
+								onkeypress="return (event.charCode !=8 && event.charCode ==0 || (event.charCode >= 48 && event.charCode <= 57))"
+								placeholder="{{ _('XXXXXXXXXXXXXXXX') }}" />
 						</div>
 
 						<div class="col-md-4 col-sm-6">
 							<label>{{ _("Card Expiry") }}</label>
 						</div>
-						<div class="col-md-8 col-sm-6">
-							<input type="text" id="card-expiry" maxlength="7" name="card-expiry" class="field"
-								placeholder="{{ _('YYYY-MM') }}" />
+						<div class="col-md-2 col-sm-3 pr-0" >
+							<input type="text" id="card-expiry-month" name="card-expiry-month" class="field"  maxlength="2"
+							onkeypress="return (event.charCode !=8 && event.charCode ==0 || (event.charCode >= 48 && event.charCode <= 57))"
```

Comment:
Please don't inline js events put it within authorizenet_checkout.js please. You can either use jquery.on or addeventlistener if you like but keep all js in one place.